[MM-59338] Migrate tooltips of "components/menu/menu.tsx" to WithTooltip

Camillarhi opened the pull request:
<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
<!--
A description of what this pull request does, as well as QA test steps (if applicable and if not already added to the Jira ticket).
-->
Migrate tooltips to use WithTooltip in "components/menu/menu.tsx"

#### Ticket Link
<!--
If applicable, please include both or either of the following links:

Fixes https://github.com/mattermost/mattermost/issues/XXX
Jira https://mattermost.atlassian.net/browse/MM-XXX
-->
Fixes https://github.com/mattermost/mattermost/issues/27554
Jira https://mattermost.atlassian.net/browse/MM-59338

#### Screenshots
<!--
If the PR includes UI changes, include screenshots/GIFs.

For an easier comparison of UI changes a table (template below) can be used.

|  before  |  after  |
|----|----|
| <insert before screenshot here> | <insert after screenshot here> |

-->

#### Release Note
<!--
Add a release note for each of the following conditions:

* Config changes (additions, deletions, updates).
* API additions—new endpoint, new response fields, or newly accepted request parameters.
* Database changes (any).
* Schema migration changes. Use the [Schema Migration Template](https://docs.google.com/document/d/18lD7N32oyMtYjFrJKwsNv8yn6Fe5QtF-eMm8nn0O8tk/edit?usp=sharing) as a starting point to capture these details as release notes. 
* Websocket additions or changes.
* Anything noteworthy to a Mattermost instance administrator (err on the side of over-communicating).
* New features and improvements, including behavioral changes, UI changes, and CLI changes.
* Bug fixes and fixes of previous known issues.
* Deprecation warnings, breaking changes, or compatibility notes.

If no release notes are required, write NONE. Use past-tense. Newlines are stripped.

Examples:

```
Added new API endpoints POST /api/v4/foo, GET api/v4/foo, and GET api/v4/foo/:foo_id.
```

```
Added a new config setting ServiceSettings.FooBar. Added a new column Foo to the Users table.
```

```
NONE
```
-->
```release-note
NONE
```

mattermost-build commented:
Hello @Camillarhi,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

M-ZubairAhmed commented:
/update-branch

M-ZubairAhmed commented on webapp/channels/src/components/menu/menu.tsx:
```diff
@@ -203,21 +201,13 @@ export function Menu(props: Props) {
-                            {props.menuButtonTooltip.text}
-                        </Tooltip>
-                    }
-                    disabled={isMenuOpen}
```
We didnt add disabled back here

M-ZubairAhmed replied:
@larkox can you please verify the disabled prop behavior do we need that?

Camillarhi replied:
@M-ZubairAhmed  I have added the disabled

M-ZubairAhmed replied:
@Camillarhi CI is still failing can you check once again

M-ZubairAhmed reviewed (changes requested):
Can you also check the failing CI

M-ZubairAhmed reviewed (changes requested):
Can you check the failing CI

M-ZubairAhmed commented:
> I am not much fond of increasing even more the `withTooltip` interface, but I guess that `disabled` is somehow nice to have. One part of me wants to keep it as:
> 
> ```
> foo = disabled ? <>{children}</> : <WithTooltip ...>{children}</WithTooltip>;
> ```
> 
> But the more I think about it, the more I see the value of the prop.

Agreed may be in the new implementation we just to this when we get the disabled prop

Camillarhi commented:
> Can you check the failing CI

The failing tests doesn't seem to come from the `menu.tsx` file

M-ZubairAhmed commented:
> > Can you check the failing CI
> 
> The failing tests doesn't seem to come from the `menu.tsx` file

The component menu is used in other places and because the structure of the menu changed and hence the snapshots where this component is used have also changed. So it is because of the change itself.

Camillarhi commented:
> > > Can you check the failing CI
> > 
> > 
> > The failing tests doesn't seem to come from the `menu.tsx` file
> 
> The component menu is used in other places and because the structure of the menu changed and hence the snapshots where this component is used have also changed. So it is because of the change itself.

How do I proceed?

Camillarhi commented:
> > > > > Can you check the failing CI
> > > > 
> > > > 
> > > > The failing tests doesn't seem to come from the `menu.tsx` file
> > > 
> > > 
> > > The component menu is used in other places and because the structure of the menu changed and hence the snapshots where this component is used have also changed. So it is because of the change itself.
> > 
> > 
> > How do I proceed?
> 
> You can update the snapshots with `npm run test:updatesnapshot`

I have updated the snapshots

M-ZubairAhmed commented:
/update-branch

M-ZubairAhmed reviewed (changes requested):
~@Camillarhi Oops sorry it looks like some more are still failing~

M-ZubairAhmed commented:
/update-branch
